l10n_nl_tax_invoice_basis/models/account_tax.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2017 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    def get_move_line_partial_domain(self, from_date, to_date, company_id):
+        res = super(AccountTax, self).get_move_line_partial_domain(
+            from_date,
+            to_date,
+            company_id
+        )
+        if not self._context.get('skip_invoice_basis_domain'):
```

Comment:
shouldn't you check here if your flag on the company is set too?

Reply:
and for stopping condition, I always recommend

```
if $stopping_condition:
    return $result
```

because then you save a level of indentation, and it's clearer what's going on imho

Reply:
so https://github.com/OCA/l10n-netherlands/pull/122#discussion_r156057093 is my only real request for change, the rest take or leave as you prefer

Reply:
With latest commit I just return if the stopping_condition is true.

The idea to keep two separate checks, one for the 'skip_invoice_basis_domain' in the context and the other for the country, is that:
if 'skip_invoice_basis_domain' is defined in the context, I skip the next check: the browse of the company and the subsequent check on its country. The intention is to save a prefetch call by the ORM, in case the context already triggered the stop condition.